src/Model/InputParser.js

```diff
@@ -0,0 +1,60 @@
+import { DEFAULT_DIVIDERS, ERROR_MESSAGES } from "../constants.js";
+
+class InputParser {
+  parse(input) {
+    if (input === "") {
+      return [0];
+    }
+
+    const { divider, numberString } = this.parseInput(input);
+    const numberArray = this.createNumberArray(numberString, divider);
+    return this.validateNumber(numberArray);
+  }
+
+  parseInput(input) {
+    if (input.startsWith("//")) {
+      return this.parseCustomDivider(input);
+    }
+
+    const defaultDividers = DEFAULT_DIVIDERS;
+    if (defaultDividers.some((div) => input.includes(div))) {
+      return { divider: defaultDividers, numberString: input };
+    }
+
+    throw new Error(ERROR_MESSAGES.NO_VALID_DIVIDER);
+  }
+
+  parseCustomDivider(input) {
+    const endIndex = input.indexOf("\\n");
+    const customDivider = input.slice(2, endIndex);
+    const numberString = input.slice(endIndex + 2);
+
+    if (numberString.length === 0) {
+      throw new Error(ERROR_MESSAGES.NO_NUMBERS_INPUT);
+    }
+
+    return { divider: [customDivider], numberString };
+  }
+
+  createNumberArray(numberString, divider) {
+    divider.forEach((divider) => {
```

Comment:
```suggestion
    dividers.forEach((divider) => {
```
`divider`를 순회해서 `divider`에 접근한다는게 살짝 어색한 부분이 있는 것 같네요🤔

`dividers`로 배열과 같이 복수형태를 띄고 있다는 걸 명시하면 어떨까요??